crates/holochain/src/core/workflow/sys_validation_workflow.rs

```diff
@@ -884,6 +901,17 @@ async fn sys_validate_record_inner(
         )
         .await;
 
+        // Check agent validity
+        if let Some(previous_action_hash) = action.prev_action() {
+            let deps = validation_dependencies.same_dht.lock();
+            // Previous action was fetched in the preceding call `fetch_previous_actions`.
+            let previous_action = deps
+                .get(previous_action_hash)
+                .and_then(|s| s.as_action())
+                .expect("previous action must have been fetched in preceding call");
```

Comment:
I would prefer returning an error here. I think a  mistake in the sys validation logic that only appears in some case we haven't figured out should only prevent validation of one op, not stop Holochain.

Reply:
I was torn, because the same call is made a couple of lines earlier and if that breaks, sys validation is bollixed. Changed it.